src/Features/CSharp/Portable/ExtractMethod/CSharpSelectionResult.ExpressionResult.cs

```diff
@@ -74,7 +75,7 @@ public override (ITypeSymbol? returnType, bool returnsByRef) GetReturnType()
                         return (regularType, returnsByRef);
 
                     if (node.Parent is CastExpressionSyntax castExpression)
-                        return (model.GetTypeInfo(castExpression).Type, returnsByRef: false);
+                        return (model.GetTypeInfo(castExpression, cancellationToken).Type, returnsByRef: false);
```

Comment:
really unfortunate cases where we're just not passing CTs around, and so potentially expensive compiler calls might block.  i'd like to make this disallowed in the IDE laye r(you have to explicitly pass a CT) but it's likely going to have a huge amount of hits to fix.

Reply:
nit: might be nice to fix the other GetTypeInfo call in this file as it looks trivial to change

Reply:
will do in followup!

Reply:
done in followup.